datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -385,7 +390,7 @@ impl LogicalPlanBuilder {
         Ok(Self::from(LogicalPlan::Prepare(Prepare {
             name,
             data_types,
-            input: Arc::new(self.plan),
+            input: Arc::clone(&self.plan),
```

Comment:
Since this method takes `self` by value, I don't think you need this clone here (or the other methods that take `self`:

```suggestion
            input: self.plan,
```

Reply:
Done